storage/redis_controller.go

```diff
@@ -160,33 +158,56 @@ func (rc *RedisController) ConnectToRedis(ctx context.Context, onConnect func(),
 	}
 
 	rc.redisUp.Store(up)
-	if up {
-		onConnect()
-	}
 
-	defer func() {
-		close(rc.reconnect)
```

Comment:
Without closing this channel, we leak the resource after ctx cancellation.

Reply:
done!